[PyTorch] Mark borrowed case as C10_LIKELY in MaybeOwned

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55940 [PyTorch] Always update op.current_dtype in TensorIteratorBase::set_output
* #55420 [PyTorch] Remove non-const TensorIterator::tensor() method
* #55258 [PyTorch] Format generated structured kernels code better
* #55351 [PyTorch] Fix const correctness for resize native functions
* #55685 [PyTorch] Avoid double indirection in MaybeOwned's borrowed state
* #55684 [PyTorch] Change MaybeOwned tests to use intrusive_ptr and Tensor
* **#55553 [PyTorch] Mark borrowed case as C10_LIKELY in MaybeOwned**
* #55419 [PyTorch] Allow copy operations on MaybeOwned

If this case isn't likely, user code would have been better off with a regular T.

Differential Revision: [D27630287](https://our.internmc.facebook.com/intern/diff/D27630287/)

swolchok committed:
[PyTorch] Mark borrowed case as C10_LIKELY in MaybeOwned

If this case isn't likely, user code would have been better off with a regular T.

Differential Revision: [D27630287](https://our.internmc.facebook.com/intern/diff/D27630287/)

[ghstack-poisoned]

imaginary-person commented:
Hello @swolchok, are `C10_LIKELY` and `C10_UNLIKELY` primarily being used to improve readability in this PR? Thanks!

swolchok committed:
Update on "[PyTorch] Mark borrowed case as C10_LIKELY in MaybeOwned"


If this case isn't likely, user code would have been better off with a regular T.

Differential Revision: [D27630287](https://our.internmc.facebook.com/intern/diff/D27630287/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Mark borrowed case as C10_LIKELY in MaybeOwned"


If this case isn't likely, user code would have been better off with a regular T.

Differential Revision: [D27630287](https://our.internmc.facebook.com/intern/diff/D27630287/)

[ghstack-poisoned]

swolchok committed:
rebase on "[PyTorch] Mark borrowed case as C10_LIKELY in MaybeOwned"


If this case isn't likely, user code would have been better off with a regular T.

Differential Revision: [D27630287](https://our.internmc.facebook.com/intern/diff/D27630287/)

[ghstack-poisoned]
